Providers/Profile.swift

```diff
@@ -606,9 +638,11 @@ public class BrowserProfile: Profile {
         var isSyncing: Bool {
             syncLock.lock()
             defer { syncLock.unlock() }
-            return !(syncReducer?.isFilled ?? true)
+            return syncDisplayState != nil && syncDisplayState! == .InProgress
```

Comment:
The adding of the sync lock to the `onStartSyncing` and `onFinishSyncing` observer methods hopefully address this.